$NetBSD: patch-ag,v 1.6 2014/05/31 07:28:57 obache Exp $

* fixes MYDOCS same as Makefile.in
* pkgsrc layout style
* prevent to overwrite with unwanted hard coded variables
* install files with wanted permissions.

--- LTmakefile.in.orig	2007-03-05 06:58:16.000000000 +0000
+++ LTmakefile.in
@@ -26,7 +26,7 @@ MYMAN1S = dpmgr.1 dptest.1 dptsv.1 crmgr
   cbtest.1 cbcodec.1 vlmgr.1 vltest.1 vltsv.1 odmgr.1 odtest.1 odidx.1 qmttest.1
 MYMAN3S = qdbm.3 depot.3 dpopen.3 curia.3 cropen.3 relic.3 hovel.3 \
   cabin.3 villa.3 vlopen.3 vista.3 odeum.3 odopen.3
-MYDOCS = spex.html spex-ja.html COPYING ChangeLog NEWS THANKS
+MYDOCS = COPYING ChangeLog NEWS THANKS
 MYPCS = qdbm.pc
 
 # Install destinations
@@ -37,20 +37,20 @@ MYLIBDIR = @libdir@
 MYBINDIR = @bindir@
 MYMAN1DIR = @mandir@/man1
 MYMAN3DIR = @mandir@/man3
-MYSHAREDIR = $(prefix)/share/$(PACKAGE)
+MYSHAREDIR = $(prefix)/share/doc/$(PACKAGE)
 MYPCDIR = @libdir@/pkgconfig
 
 # Building binaries
 LIBTOOL = libtool
-CC = gcc
-CFLAGS = -I. -I$(MYHEADDIR) -I$(HOME)/include -I/usr/local/include @MYDEFS@ \
+CC = @CC@
+CFLAGS = -I. -I$(MYHEADDIR) @CFLAGS@ @MYDEFS@ \
   -D_XOPEN_SOURCE_EXTENDED=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__=1 -D_HPUX_SOURCE=1 \
   -D_POSIX_MAPPED_FILES=1 -D_POSIX_SYNCHRONIZED_IO=1 \
-  -DPIC=1 -D_THREAD_SAFE=1 -D_REENTRANT=1 -DNDEBUG -O3
-LD = gcc
-LIBLDFLAGS = -rpath $(MYLIBDIR) -R $(MYLIBDIR) \
+  -DPIC=1 -D_THREAD_SAFE=1 ${PTHREAD_CFLAGS} -DNDEBUG
+LD = @CC@
+LIBLDFLAGS = @LDFLAGS@ -rpath $(MYLIBDIR) -R $(MYLIBDIR) \
   -version-info $$(($(LIBVER)+$(LIBREV))):0:$(LIBREV) @LIBS@
-LDFLAGS = -rpath $(MYLIBDIR) -R $(MYLIBDIR) -L. -lqdbm @LIBS@
+LDFLAGS = @LDFLAGS@ -rpath $(MYLIBDIR) -R $(MYLIBDIR) -L. -lqdbm @LIBS@
 INSTALL = install
 MKDIR = mkdir -p
 CP = cp -rf
@@ -89,17 +89,17 @@ clean :
 
 
 install :
-	$(LIBTOOL) --mode=install $(INSTALL) $(MYHEADS) $(MYHEADDIR)
-	$(LIBTOOL) --mode=install $(INSTALL) $(MYLIBS) $(MYLIBDIR)
-	$(LIBTOOL) --mode=install $(INSTALL) $(MYBINS) $(MYBINDIR)
-	$(MKDIR) $(MYMAN1DIR)
-	cd man && $(CP) $(MYMAN1S) $(MYMAN1DIR)
-	$(MKDIR) $(MYMAN3DIR)
-	cd man && $(CP) $(MYMAN3S) $(MYMAN3DIR)
-	$(MKDIR) $(MYSHAREDIR)
-	$(CP) $(MYDOCS) $(MYSHAREDIR)
-	$(MKDIR) $(MYPCDIR)
-	$(CP) $(MYPCS) $(MYPCDIR)
+	${BSD_INSTALL_DATA} $(MYHEADS) $(DESTDIR)$(MYHEADDIR)
+	$(LIBTOOL) --mode=install $(INSTALL) $(MYLIBS) $(DESTDIR)$(MYLIBDIR)
+	$(LIBTOOL) --mode=install $(INSTALL) $(MYBINS) $(DESTDIR)$(MYBINDIR)
+	${BSD_INSTALL_MAN_DIR} $(DESTDIR)$(MYMAN1DIR)
+	cd man && ${BSD_INSTALL_MAN} $(MYMAN1S) $(DESTDIR)$(MYMAN1DIR)
+	${BSD_INSTALL_MAN_DIR} $(DESTDIR)$(MYMAN3DIR)
+	cd man && ${BSD_INSTALL_MAN} $(MYMAN3S) $(DESTDIR)$(MYMAN3DIR)
+	${BSD_INSTALL_DATA_DIR} $(DESTDIR)$(MYSHAREDIR)
+	${BSD_INSTALL_DATA} $(MYDOCS) $(DESTDIR)$(MYSHAREDIR)
+	${BSD_INSTALL_DATA_DIR} $(DESTDIR)$(MYPCDIR)
+	${BSD_INSTALL_DATA} $(MYPCS) $(DESTDIR)$(MYPCDIR)
 	@printf '\n'
 	@printf '#================================================================\n'
 	@printf '# Thanks for using QDBM.\n'
